lib/internal/watchdog.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const {
+  JsSigintWatchdog,
+  None,
+  FromIdle,
+  FromInterrupt
+} = internalBinding('watchdog');
+
+class SigintWatchdog extends JsSigintWatchdog {
+  _started = false;
+  _effective = false;
+  _onNewListener = (eve) => {
+    if (eve === 'SIGINT' && this._effective) {
+      super.stop();
+      this._effective = false;
+    }
+  };
+  _onRemoveListener = (eve) => {
+    if (eve === 'SIGINT' && process.listenerCount('SIGINT') === 0 &&
+        !this._effective) {
+      super.start();
+      this._effective = true;
+    }
+  }
+
+  start() {
+    if (this._started) {
+      return;
+    }
+    this._started = true;
+    /**
+     * Prepend sigint newListener to remove stop watchdog before signal wrap
+     * been activated. Also make sigint removeListener been ran after signal
+     * wrap been stopped.
+     */
```

Comment:
Can you use `//` comments? Thanks.

Reply:
Maybe we could enable eslint rule `multiline-comment-style` to enforce this style?